minio/config/hook-helper.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+# This helper optimizes minio hooks by:
+# * extracting toml variables to bash
+# * creating helper methods
+# * keeping secrets out: https://github.com/habitat-sh/habitat/issues/6076
+#
+# Expected be sourced by all hooks at very early stage
+
+standalone="{{ cfg.standalone }}"
+expected="{{ cfg.expected }}"
+use_ssl="{{ cfg.use_ssl }}"
+ssl_cert_pw="{{ cfg.ssl_cert_pw }}"
+
+# This hook is used for caching last known good configuration
+run_good_hook="{{pkg.svc_static_path}}/run-good"
+
+# This meta hook is used for restart detection
+meta_restart_hook="{{pkg.svc_config_path}}/restart-hook.txt"
+
+# For aws cli
+export AWS_ACCESS_KEY_ID="{{cfg.key_id}}"
+export AWS_SECRET_ACCESS_KEY="{{cfg.secret_key}}"
+
+# For minio server
+export MINIO_ACCESS_KEY="{{cfg.key_id}}"
+export MINIO_SECRET_KEY="{{cfg.secret_key}}"
+
+if [ -n "$ssl_cert_pw" ]; then
+    export MINIO_CERT_PASSWD="$ssl_cert_pw"
+fi
+
+# Generate Expected Members
+# {{#eachAlive svc.members as |member|}}
+svc_members="$svc_members http://{{member.sys.hostname}}{{../pkg.svc_data_path}}"
+# {{/eachAlive}}
+svc_count=$(echo "$svc_members" | wc -w)
+
+# Generate Explicit Members
+# {{#each cfg.members as |member|}}
+explicit_members="$explicit_members {{member}}"
+# {{/each}}
+explicit_count=$(echo "$explicit_members" | wc -w)
+
+# Helper methods
+
+# Can't calculate PID on the early stage, so defer it by method
+minio_pid() {
+    cat "{{pkg.svc_pid_file}}"
+}
+
+# Copy certificates from ring
+copy_certs() {
+    cert_dir="{{pkg.svc_config_path}}/certs"
+    priv_key="{{pkg.svc_files_path}}/private.key"
+    pub_key="{{pkg.svc_files_path}}/public.crt"
+
+    mkdir -p $cert_dir
+
+    cp $priv_key $cert_dir
+    cp $pub_key $cert_dir
+}
+
+# TODO: Remove or refactor this method
```

Comment:
@predominant I can't fix this because of issue. Before do post-run actions (like create bucket) I need to be sure that minio is up and running, but initialization itself takes time. I had to TEMPORARY add splay delay for reliable start.
I also don't know how to exclude this particular sleep-check.
Do we have workarounds here?

Reply:
Personally I don't think a small sleep like this is dangerous because it will always exit.

What I think is important is that the warning comment here be stronger like so:

```
# TODO: Re-evaluate when hooks can run on their own thread
# Sleeps in hooks that are not the run hook
# are not a good practice because all hooks run in the Supervisor's
# main thread. In the future, this may change.
# See also: https://github.com/habitat-sh/habitat/issues/2364
# Wait so elasticsearch can start
```

We want to be careful that others in core plans do not adopt this practice without fully understanding what they are doing is unsafe and not a good practice.

I think if we have something like that, that we can do the practical thing and get this merged so that it can be used.


In terms of the Travis build.... I'm not exactly sure? I was able to get the travis build to pass, and the only difference I see between my code and yours, it that I don't have spaces in my handlebars syntax. See: https://github.com/habitat-sh/core-plans/pull/2660/files